tests/src/main/java/org/hibernate/beanvalidation/tck/tests/constraints/containerelement/ContainerElementConstraintMapKeyTest.java

```diff
@@ -0,0 +1,309 @@
+								.parameter( "mapParameter", 0 )
+								.containerElement( "<map key>", true, null, null, Map.class, 0 )
+						),
+				violationWith()
+						.constraintType( NotNull.class )
+						.propertyPath( pathWith()
+								.method( "setValues" )
+								.parameter( "mapParameter", 0 )
+								.containerElement( "<map key>", true, null, null, Map.class, 0 )
+						)
+		);
+	}
+
+	@Test
+	@SpecAssertion(section = Sections.CONSTRAINTDECLARATIONVALIDATIONPROCESS_CONTAINERELEMENTCONSTRAINTS, id = "a")
+	@SpecAssertion(section = Sections.CONSTRAINTDECLARATIONVALIDATIONPROCESS_CONTAINERELEMENTCONSTRAINTS, id = "b")
+	@SpecAssertion(section = Sections.CONSTRAINTDECLARATIONVALIDATIONPROCESS_CONTAINERELEMENTCONSTRAINTS, id = "c")
+	public void constructor_parameter_constraint_provided_on_key_type_parameter_of_a_map_gets_validated() throws Exception {
+		Constructor<TypeWithMap6> constructor = TypeWithMap6.class.getDeclaredConstructor( Map.class );
+
+		Map<String, String> parameter = new HashMap<>();
+		parameter.put( "first", "First" );
+		parameter.put( "", "Second" );
+		parameter.put( null, "Third" );
+		Object[] values = new Object[] { parameter };
+
+		Set<ConstraintViolation<TypeWithMap6>> constraintViolations = getExecutableValidator().validateConstructorParameters(
+				constructor,
+				values
+		);
+
+		assertThat( constraintViolations ).containsOnlyViolations(
+				violationWith()
+						.constraintType( NotBlank.class )
+						.propertyPath( pathWith()
+								.constructor( TypeWithMap6.class )
+								.parameter( "mapParameter", 0 )
+								.containerElement( "<map key>", true, "", null, Map.class, 0 )
+						),
+				violationWith()
+						.constraintType( NotBlank.class )
+						.propertyPath( pathWith()
+								.constructor( TypeWithMap6.class )
+								.parameter( "mapParameter", 0 )
+								.containerElement( "<map key>", true, null, null, Map.class, 0 )
+						),
+				violationWith()
+						.constraintType( NotNull.class )
+						.propertyPath( pathWith()
+								.constructor( TypeWithMap6.class )
+								.parameter( "mapParameter", 0 )
+								.containerElement( "<map key>", true, null, null, Map.class, 0 )
+						)
+		);
+	}
+
+	private static class TypeWithMap1 {
+
+		private Map<@NotBlank String, String> nameMap;
+	}
+
+	private static class TypeWithMap2 {
+
+		@NotNull
+		private Map<@NotBlank String, String> nameMap;
+	}
+
+	private static class TypeWithMap3 {
+
+		private Map<String, String> stringMap;
+
+		@SuppressWarnings("unused")
+		public Map<@NotNull @NotBlank String, String> getStringMap() {
+			return stringMap;
+		}
+	}
+
+	private static class TypeWithMap4 {
+
+		private Map<String, String> stringMap;
+
+		@SuppressWarnings("unused")
+		public Map<@NotNull @NotBlank String, String> returnStringMap() {
+			return stringMap;
+		}
+	}
+
+	private static class TypeWithMap5 {
+
+		@SuppressWarnings("unused")
+		public void setValues(@Valid Map<@NotNull @NotBlank String, String> mapParameter) {
```

Comment:
Same question about using the legacy style. Is it even needed here at all given that this test is about the map key?